Fix/ci paths 2

timgetz opened the pull request:
feature/ARCWELL-235 increase coverage of server test suite (#204)
    - Adding a cache action for postgres image
    - Added cron job to run nightly, but only push image if manifests have changed
    - Testing pulling the cached postgres image - ghcr.io/arcweb/arcwell/postgres:16-bookworm
    - Cleaned up admin & server -ci.yml files using the built in caching configuration
    - Removed the unnessary manual caching and artifact creation
    - Fixed server & admin ci configuration
    - Commented out failing resource server test (Covered in a bug ticket)
    - Temporarily commented out Admin server execution until ARCWELL-387 is implemented.
    - Added CI Status badges to the README

timgetz committed:
fix/ci-paths
- Fixing ci paths

timgetz committed:
fix/ci-paths
- Fixing ci paths

timgetz committed:
fix/ci-paths
- Fixing ci paths

timgetz committed:
fix/ci-paths
- Fixing ci paths

timgetz committed:
fix/ci-paths
- Fixing ci paths

timgetz committed:
fix/ci-paths
- Fixing ci paths

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls
- Re-added Admin tests

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls
- Re-added Admin tests

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls
- Re-added Admin tests

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls
- Re-added Admin tests
- Re-added check for changes back

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls
- Re-added Admin tests
- Re-added check for changes back

timgetz committed:
fix/ci-paths
- Fixed ci paths
- Fixed server authentication in tests
- Added api/v1 to base url
- Fixed docker ci ymls
- Fixed server ci configuration
- Consolidated compose ci ymls
- Re-added Admin tests
- Re-added check for changes back
- Adding list-files to filter

timgetz committed:
fix/ci-paths
- Cleaned up admin & server -ci.yml files using the built in caching configuration
- Removed the unnessary manual caching and artifact creation
- Fixed server & admin ci configuration
Commented out failing resource server test (Covered in a bug ticket)
- Temporarily commented out Admin server execution until ARCWELL-387 is implemented.

timgetz committed:
fix/ci-paths
- Cleaned up admin & server -ci.yml files using the built in caching configuration
- Removed the unnessary manual caching and artifact creation
- Fixed server & admin ci configuration
- Commented out failing resource server test (Covered in a bug ticket)
- Temporarily commented out Admin server execution until ARCWELL-387 is implemented.
- Added CI Status badges to the README

timgetz committed:
fix/ci-paths-2
- Adding a cache for postgres image

timgetz committed:
feature/postgres-image-cache
- Adding a cache action for postgres image
- Testing pulling the cached postgres image - ghcr.io/arcweb/arcwell/postgres:16-bookworm

mccolin commented on .github/workflows/admin-ci.yml:
```diff
@@ -91,58 +101,19 @@ jobs:
             echo "Health check failed after 10 attempts. Exiting."
+            docker compose logs server
             exit 1
           fi
```
Can we rename all of this logic up to be part of the step before where we launch the services? And rename this from health checking to confirming server is up? This doesn't have to be two separate job steps.

Also -- can we take advantage of some Dockerfile for Compose File settings to confirm a service is up without customer scripts? May be simpler.

Looks like because we already have a healthcheck specified in the Dockerfile, we can make the CI compose file use `depends_on` with `condition` and `service_healthy`: https://docs.docker.com/reference/compose-file/services/#depends_on

timgetz replied:
As we agreed, we'll make this an enhancmeent and I created a ticket for it
https://arcweb.atlassian.net/browse/ARCWELL-459

mccolin commented on .github/workflows/admin-ci.yml:
```diff
@@ -91,58 +101,19 @@ jobs:
+      # TODO: Temporarily commenting this out until ARCWELL-387 "Implement Admin unit testing is completed"
+#      - name: Test Server
+#        run: |
+#          docker compose run admin ng test --watch=false
```
Should this eventually be a `docker compose exec` instead of `run`?

timgetz replied:
Fixed

mccolin commented on .github/workflows/push-postgres-ci.yml:
```diff
@@ -0,0 +1,54 @@
+on:
+  schedule:
+    - cron: '0 0 * * *'  # Daily at midnight UTC
+  workflow_dispatch:     # Allows manual triggering
```
Does this have to run nightly or only once?

timgetz replied:
As I commented in slack, it can be whatever we want and/or both.

timgetz committed:
feature/postgres-image-cache
- Adding a cache action for postgres image
- Testing pulling the cached postgres image - ghcr.io/arcweb/arcwell/postgres:16-bookworm
- Added check to only push new image if there are changes
- Resolved code review comments

timgetz committed:
feature/postgres-image-cache
- Adding a cache action for postgres image
- Testing pulling the cached postgres image - ghcr.io/arcweb/arcwell/postgres:16-bookworm
- Added check to only push new image if there are changes
- Resolved code review comments
